### PR TITLE
Implement login functionality

### PR DESCRIPTION
**Pull request by kelvinndmo**
- This pull request adds the ability for patients to login.
- Generates a JWT token if the credentials are ok.

### CODE REVIEW

File: dtos/user.py [hunk 1]
```diff
@@ -41,3 +41,21 @@ class UpdateUserReponse(BaseModel):
     """
 
     data: User
+
+
+class LoginPayload(BaseModel):
```
Comment (wilsonangara) [inline dtos/user.py:46]: @kelvinndmo what is the difference between 1 and 2 new lines? or is it just the convention in python?

**Review by wilsonangara — APPROVED** [review]
LGTM

File: repositories/credentials.py [hunk 2]
```diff
@@ -13,13 +10,35 @@
...
+    phone_number: str, credential: CredentialsModel, message: str = ""
+):
+    """
+    Generate otp and send message
+    """
+    otp_value = random.randint(100000, 999999)
+    credential.value = get_hashed_password(str(otp_value))
+    credential.alg = "otp"
+    credential.is_hashed = True
+    plivo_client.messages.create(
```
Comment (linerocks) [inline repositories/credentials.py:27]: is this a sync or async process?

File: repositories/credentials.py [hunk 3]
```diff
@@ -28,27 +47,18 @@ async def create_credentials(payload: CreateCredentialPayload) -> CredentialResp
...
         credentials.alg = "bcrypt"
         credentials.is_hashed = True
     if payload.credential_type == CredentialTypeEnum.PHONE_OTP:
-        # generate otp, we don
-        password_value = random.randint(100000, 999999)
-        credentials.value = bcrypt.hashpw(
-            str(password_value).encode("utf-8"),
-            bcrypt.gensalt(),
+        credentials = await generate_otp_and_send_message(
+            credential=credentials, phone_number=user.phone_number
```
Comment (linerocks) [inline repositories/credentials.py:55]: I think this is confusing. The variable has a plural name, but it's meant to be singular. I suggest making it singular, just `credential`.

File: repositories/user.py [hunk 4]
```diff
@@ -63,9 +77,40 @@ async def update_user(user_id: str, payload: UpdateUserPayload) -> UserModel:
...
+    """
+    Check if phone number exists
+    """
+    phone_exists = await UserModel.find(
+        UserModel.phone_number == phone_number
+    ).to_list()
+    return len(phone_exists) > 0
+
+
+async def contact_details_temp_change_by_user_id_credential_id(
```
Comment (linerocks) [inline repositories/user.py:107]: need to find better name :D

File: routes/user.py [hunk 5]
```diff
@@ -61,7 +65,93 @@ async def update_user(user_id: str, payload: UpdateUserPayload):
...
+        await UserRepository.contact_details_temp_change_by_user_id_credential_id(
+            user_id, payload.credential_id
+        )
+    )
+    if not temp_details:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="We couldn't verify your update request",
+        )
+    user = await UserRepository.update_user(
```
Comment (linerocks) [inline routes/user.py:145]: can this be included in asyncio below too?

**Comment by kelvinndmo** [comment]
ack and fixed @linerocks

**Pull request merged**